beets/dbcore/db.py

```diff
@@ -290,10 +290,11 @@ def __delitem__(self, key):
         if key in self._values_flex:  # Flexible.
             del self._values_flex[key]
             self._dirty.add(key)  # Mark for dropping on store.
+        elif key in self._fields:  # Fixed
+            setattr(self, key, self._type(key).null)
+            self._dirty.add(key)  # Mark for dropping on store.
```

Comment:
This comment seems slightly inaccurate for this line: maybe “mark for updating” or “mark for setting to null.”

Actually, come to think of it, I’m not sure this dirtying line is necessary. We need it for flexible attributes because we change the underlying dict directly, but `setattr` should trigger the normal field modification mechanism, which automatically sets the dirty flag. Would you mind giving this a try without this line?

Reply:
You are correct. It wasn't needed. I've updated the PR.